### PR TITLE
Validate MDX links

### PR DESCRIPTION
**Pull request by hasparus**
### Background

[Thread on Slack](https://northstar-e5s5760.slack.com/archives/C07DYCW4NUD/p1733416767816089)

### Description

Used `next-validate-link` to add a CI check that ensures all links lead to pages that exist.

### CODE REVIEW

File: .github/workflows/validate-mdx.links.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          fetch-depth: 2
+
+      - name: setup environment
+        uses: ./.github/actions/setup
+        with:
+          actor: validate-mdx-links
+          cacheTurbo: false
+
+      - name: Validate Docs MDX Links
+        run: ./scripts/validate-mdx-links.ts --cwd packages/web/docs
```
Comment (hasparus) [inline .github/workflows/validate-mdx.links.yaml:26]: Not sure if this should be separate workflow or part of `lint`. I'll leave this decision to someone else.

File: scripts/validate-mdx-links.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env node --experimental-strip-types --no-warnings=ExperimentalWarning
```
Comment (hasparus) [inline scripts/validate-mdx-links.ts:1]: I'm never writing a Bash script again.
Would prefer `bun`, but `node --experimental-strip-types` is pretty great.

File: scripts/validate-mdx-links.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+process.chdir(args.values.cwd);
+
+const files = globSync(args.values.files);
+
+if (files.length === 0) {
+  console.error('No files found. Please pass the --cwd or navigate to the proper directory.');
+  process.exit(1);
+} else {
+  console.log(`Found ${files.length} markdown files to validate.\n`);
+}
```
Comment (dimaMachina) [inline scripts/validate-mdx-links.ts:32]:
```suggestion
}
console.log(`Found ${files.length} markdown files to validate.\n`);
```

unneeded `else`

**Review by dimaMachina — APPROVED** [review]

File: scripts/validate-mdx-links.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { globSync } from 'node:fs';
+import { stat } from 'node:fs/promises';
+import { dirname, resolve } from 'node:path';
+import { parseArgs } from 'node:util';
+import { printErrors, scanURLs, validateFiles, type DetectedError } from 'next-validate-link';
+
+const args = parseArgs({
+  options: {
+    cwd: { type: 'string', default: process.cwd() },
+    files: { type: 'string', default: 'src/content/**/*.mdx' },
```
Comment (kamilkisiela) [inline scripts/validate-mdx-links.ts:19]: why it's `src/content`? Can you add a comment in code?

Comment (hasparus) [reply scripts/validate-mdx-links.ts:19]: I think this part will need to be changed now as Dima moved product updates to app, so we have to check `src/{app,content}`. I'll test it after we get ecosystem page merged.

**Pull request merged**